src/api/java/baritone/api/utils/ExampleBaritoneControl.java

```diff
@@ -688,18 +691,23 @@ private Goal parseGoal(String[] params) {
         Goal goal;
         try {
             BetterBlockPos playerFeet = ctx.playerFeet();
-            switch (params.length) {
+
+            int length = params.length - 1; // length has to be smaller when a dimension parameter is added
+            if(params.length < 1 || (isNumeric(params[params.length - 1]) || params[params.length - 1].startsWith("~"))){
+                length = params.length;
+            }
+            switch (length) {
                 case 0:
                     goal = new GoalBlock(playerFeet);
                     break;
                 case 1:
-                    goal = new GoalYLevel(parseOrDefault(params[0], playerFeet.y));
+                    goal = new GoalYLevel(parseOrDefault(params[0], playerFeet.y, 1));
                     break;
                 case 2:
-                    goal = new GoalXZ(parseOrDefault(params[0], playerFeet.x), parseOrDefault(params[1], playerFeet.z));
+                    goal = new GoalXZ(parseOrDefault(params[0], playerFeet.x, calculateDimensionFactor(ctx.world().provider.getDimensionType(), getDimensionByName(params[params.length - 1]))), parseOrDefault(params[1], playerFeet.z, calculateDimensionFactor(ctx.world().provider.getDimensionType(), getDimensionByName(params[params.length - 1]))));
```

Comment:
if `from` is always `ctx.world().provider.getDimensionType()` then shouldn't `calculateDimensionFactor` just take in the `to`, and it can do `ctx.world().provider.getDimensionType()` within the function?

Reply:
yea true. I thought other functions might use the function too but I think all helper functions are just used in that file where they are defined. So I moght as well make them private